### PR TITLE
Reader: add combined card component

### PR DESCRIPTION
**Pull request by bluefuton**
Add a new `reader-combined-card` block to show a number of consecutive posts from a single site in Reader streams.

The combined card should look like this:

combined-card-wide

And on narrower viewports:

combined-card-narrow

### To test

View the new component in Devdocs:

http://calypso.localhost:3000/devdocs/blocks/reader-combined-card

It will be added to streams in a later PR.

### CODE REVIEW

**Comment by bluefuton** [comment]
Todo:
- [ ] author name
- [ ] time since publication
- [ ] timescale of posts ("5 posts in x hours") - might make sense in separate PR
- [ ] post thumbnail
- [ ] style header
- [ ] style posts

**Comment by bluefuton** [comment]
Still todo (CSS-y):
- [ ] Clamp excerpt to one line
- [ ] Add `long-content-fade` on title and excerpt
- [ ] Lay out byline 
- [ ] Fix spacing for posts without a featured image
- [ ] Fix spacing between posts

cc @raoulwegat 

Other bits: 
- [ ] Handle video thumbnails

File: client/blocks/reader-avatar/index.jsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import SiteIcon from 'blocks/site-icon';
 import { localize } from 'i18n-calypso';
 import classnames from 'classnames';
 
-const ReaderAvatar = ( { author, siteIcon, feedIcon, siteUrl, preferGravatar = false, showPlaceholder = false } ) => {
+const ReaderAvatar = ( { author, siteIcon, feedIcon, siteUrl, siteIconSize = 96, preferGravatar = false, showPlaceholder = false } ) => {
```
Comment (bluefuton) [inline client/blocks/reader-avatar/index.jsx]: Added `siteIconSize` param so we can have a smaller image for the combined card header.

File: client/blocks/reader-featured-image/index.jsx [hunk 2]
```diff
@@ -10,15 +10,13 @@ import { noop } from 'lodash';
 import cssSafeUrl from 'lib/css-safe-url';
 import resizeImageUrl from 'lib/resize-image-url';
 
-const FEATURED_IMAGE_WIDTH = 250;
-
-const ReaderFeaturedImage = ( { imageUrl, href, children, onClick } ) => {
+const ReaderFeaturedImage = ( { imageUrl, imageWidth, href, children, onClick } ) => {
```
Comment (bluefuton) [inline client/blocks/reader-featured-image/index.jsx]: Allow an `imageWidth` to be provided (we want smaller images for combined card posts).

File: client/blocks/reader-post-card/docs/example.jsx [hunk 3]
```diff
@@ -7,139 +7,16 @@ import React from 'react';
...
-			discussion: {
-				comment_count: 99
-			},
-			date: '1976-09-15T10:12:00Z',
-			site: 'cats.wordpress.com',
-			better_excerpt_no_html: 'Every four years, the Get Out the Vote campaign invites graphic designers to make posters that rally US voters to go to the polls. Here, 14 posters that rock…'
-		}
-	},
-];
+import { posts, site } from './fixtures';
```
Comment (bluefuton) [inline client/blocks/reader-post-card/docs/example.jsx]: Move fixtures out to a separate file so they can be reused.

**Comment by samouri** [comment]
I haven't had a chance to go through all of the code yet, but heres something I noticed while playing around with it:

When tabbing through the links in the combined card the titles become bright blue: 
screen shot 2017-03-06 at 11 13 19 am

I have no idea why

**Comment by bluefuton** [comment]
> When tabbing through the links in the combined card the titles become bright blue

Looks like a stray `:active` state. Cheers, I'll have a look!

File: client/blocks/reader-combined-card/index.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	const streamUrl = getStreamUrl( feedId, siteId );
+	const siteName = siteNameFromSiteAndPost( site, posts[ 0 ] );
+
+	return (
+		<Card className="reader-combined-card">
+			<header className="reader-combined-card__header">
+				<ReaderAvatar
+					siteIcon={ siteIcon }
+					feedIcon={ feedIcon }
+					author={ null }
```
Comment (samouri) [inline client/blocks/reader-combined-card/index.jsx]: why does author = null?

Comment (bluefuton) [inline client/blocks/reader-combined-card/index.jsx]: We don't have an author :)

In the combined card header, we only ever show the site or feed icon.

File: client/blocks/reader-combined-card/post.jsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+		// if the click has modifier or was not primary, ignore it
+		if ( event.button > 0 || event.metaKey || event.controlKey || event.shiftKey || event.altKey ) {
+			if ( closest( event.target, '.reader-combined-card__post-title-link', true, rootNode ) ) {
+				stats.recordPermalinkClick( 'card_title_with_modifier', this.props.post );
+			}
+			return;
+		}
+
+		if ( closest( event.target, '.should-scroll', true, rootNode ) ) {
```
Comment (samouri) [inline client/blocks/reader-combined-card/post.jsx]: what does this if and the if below it do?

File: client/blocks/reader-combined-card/post.jsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+import * as stats from 'reader/stats';
+
+class ReaderCombinedCardPost extends React.Component {
+	static propTypes = {
+		post: React.PropTypes.object.isRequired,
+		streamUrl: React.PropTypes.string,
+		onClick: React.PropTypes.func,
+	};
+
+	propagateCardClick = () => {
```
Comment (samouri) [inline client/blocks/reader-combined-card/post.jsx]: can we ever have discover cards in here?
If not maybe remove the func and put `this.props.onClick( this.props.post ) ` within handleCardClick.

File: client/blocks/reader-combined-card/post.jsx (hunk 5 above)
Comment (bluefuton) [inline client/blocks/reader-combined-card/post.jsx]: Both present in the post card too.

`.should-scroll` - I'll defer to @blowery to explain. Still needed I believe.

The one below ignores any clicks on an element with `.ignore-click` class applied. In the combined card, the visit link/author/timestamp use this to allow them to do something different to the card click handler (e.g. open an external site).

File: client/blocks/reader-combined-card/style.scss [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+	color: $gray;
+	margin-bottom: 0;
+	line-height: 1em;
+}
+
+/* Posts */
+.reader-combined-card__post-details {
+	font-family: $serif;
+	width: 100%;
+
```
Comment (samouri) [inline client/blocks/reader-combined-card/style.scss]: extra newline

File: client/blocks/reader-combined-card/post.jsx (hunk 6 above)
Comment (bluefuton) [inline client/blocks/reader-combined-card/post.jsx]: Yep - Discover could potentially be one of the sites that's bundled up (if you're subscribed to few very feeds).

File: client/blocks/reader-combined-card/post.jsx [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+				<div className="reader-combined-card__featured-image-wrapper">
+					{ featuredAsset }
+				</div>
+				<div className="reader-combined-card__post-details">
+					<AutoDirection>
+						<h1 className="reader-combined-card__post-title">
+							<a className="reader-combined-card__post-title-link" href={ post.URL }>{ post.title }</a>
+						</h1>
+					</AutoDirection>
+					<ReaderExcerpt post={ post } isDiscover={ false } />
```
Comment (samouri) [inline client/blocks/reader-combined-card/post.jsx]: should `isDiscover` always be false?

Comment (bluefuton) [inline client/blocks/reader-combined-card/post.jsx]: Nope - good spot! The combined card should probably accept this as a prop and pass it down to the post. I'll fix.

Comment (bluefuton) [inline client/blocks/reader-combined-card/post.jsx]: Fixed in af3ad59 👍

File: client/blocks/reader-combined-card/post.jsx (hunk 5 above)
Comment (blowery) [inline client/blocks/reader-combined-card/post.jsx]: I'm pretty confident `should-scroll` is no longer used... 

Good explanation of `ignore-click` and why we still require it

Comment (bluefuton) [inline client/blocks/reader-combined-card/post.jsx]: Cool - thanks @blowery! I'll drop `.should-scroll`.

**Comment by samouri** [comment]
giving this a final test

**Pull request merged**